pkg/controller/installation/core_controller.go

```diff
@@ -742,26 +748,41 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 	terminating := (instance.DeletionTimestamp != nil)
 	preDefaultPatchFrom := client.MergeFrom(instance.DeepCopy())
 
+	// Changes for updating installation status conditions
+	if request.Name == "calico" && request.Namespace == "" {
+		ts := &operator.TigeraStatus{ObjectMeta: metav1.ObjectMeta{Name: "calico"}}
+		err := r.client.Get(context.TODO(), types.NamespacedName{Name: "calico"}, ts)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		r.updateInstallationStatus(instance, ts.Status.Conditions)
+		if err := r.client.Status().Update(ctx, instance); err != nil {
+			log.WithValues("reason", err).Info("Failed to create Installation status conditions.")
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{}, nil
+	}
+
 	// Mark CR found so we can report converter problems via tigerastatus
 	r.status.OnCRFound()
 
 	if !r.migrationChecked {
 		// update Installation resource with existing install if it exists.
 		nc, err := convert.NeedsConversion(ctx, r.client)
 		if err != nil {
-			r.SetDegraded("Error checking for existing installation", err, reqLogger)
+			r.SetDegraded(operator.ResourceUpdateError, "Error checking for existing installation", err, reqLogger)
```

Comment:
NeedsConversion can throw an error for different reasons: NotFound, parsing, env missing. Therefore you should modify the func such that it also returns the Reason.

Reply:
Done